frontend/src/stores/contracts.ts

```diff
@@ -177,12 +177,25 @@ export const useContractsStore = defineStore('contractsStore', () => {
       const contract = deployedContracts.value.find(
         (c) => c.contractId === localContractId,
       );
-      const result = await $jsonRpc?.callContractFunction({
-        userAccount,
-        contractAddress: contract?.address || '',
-        method,
-        params,
-      });
+
+      if (!accountsStore.currentPrivateKey) {
+        return false;
+      }
+
+      const methodParamsAsString = JSON.stringify(params)
+      const data = toRlp([
+        toHex(method),
+        toHex(methodParamsAsString)
+      ])
+      
+      const to = contract?.address as Address ?? null;
+      const signed = await signTransaction(
```

Comment:
This uses the internal Viem method to sign the transaction. Here is this function implementation: https://github.com/yeagerai/genlayer-simulator/pull/315/files#diff-fa9662a1666b1c93a605e8a50d112a7884f29d058380bff14cd13366fa9b9cc4R5